roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/StreetSection.java

```diff
@@ -47,21 +97,56 @@ public IStreetConnector getPreviousStreetConnector() {
 
     @Override
     public Map<ICar, Double> getCarPositions() {
-        return carPositions;
+        return Collections.unmodifiableMap(carPositions);
     }
 
+    @Override
     public void updateAllCarsPositions() {
-
+        final double currentTime = getCurrentTime();
+
+        // Updating positions for all cars.
+        ICar previousCar = null;
+        for (ICar currentCar : carQueue) {
+            final double carLastUpdateTime = currentCar.getLastUpdateTime();
+            final double carSpeed = currentCar.getDriverBehaviour().getSpeed();
+            final double carPosition = getCarPositions().getOrDefault(currentCar, INITIAL_CAR_POSITION);
+
+            // Calculate distance to next car / end of street section based on distributed driver behaviour values.
+            final double distanceToNextCar = 0; //TODO finish.
+            currentCar.getDriverBehaviour().getMinDistanceToNextCar();
+            currentCar.getDriverBehaviour().getMaxDistanceToNextCar();
+
+            final double maxTheoreticallyPossiblePositionValue = getMaxPossibleCarPosition(
+                getLength(),
+                distanceToNextCar,
+                getCarPositions().get(previousCar),
+                previousCar
```

Comment:
format ok?

Reply:
ok now i get about which format you were talking. if it will be written in a single line it will take more than 120 chars in a line which is more than regular ide can display at full hd resolution (you will how to horizontally scroll which is disgusting).